Improve loading placeholder animation performance

PVince81 opened the pull request:
Fixes https://github.com/nextcloud/spreed/issues/4553

Replaced the SVG gradient animation with a CSS opacity animation that fades between two lists, one with the regular gradient and one with the reversed one.

See ticket for details about analysis.

The fix reduces the CPU load for me from 54% to ~20 %

PVince81 commented:
If that feels a bit over the top, I had another local version where I only pulsed a single list. The CPU usage was almost the same, so if we want we could still stick with the gradient+opacity thing.

ma12-co reviewed (approved):
Nice, the only thing is that for me the placeholders are aligned to the left instead of centered!

skjnldsv commented on src/components/LoadingPlaceholder.vue:
```diff
@@ -20,51 +20,50 @@
-							repeatCount="indefinite" />
-					</stop>
+				<linearGradient id="placeholder-gradient-regular">
+					<stop offset="0%" :stop-color="light" />
```
You can't use variables there?

nickvergessen replied:
for `0%` ?

PVince81 replied:
adjusted

skjnldsv replied:
Lol no, for `stop-color="var(--color-blabla)"`

skjnldsv replied:
Ah, ignore my comments, I think I misunderstood the code :thinking:

PVince81 commented:
fixed centering and squashed.

I've compared the animation with the old one side by side and couldn't see any significant difference, they look visually the same.

Note: it seems the avatar size and bar alignment doesn't match whatever the actual list has. I attempted to adjust it but failed as the alignment changes with different zoom sizes... If we want alignment identical to the message list we might need to rework the svg markup and use HTML / CSS for alignment instead of CSS calculations. This is out of scope for the current PR. (no quick win sadly)

nickvergessen commented on src/components/LoadingPlaceholder.vue:
```diff
@@ -18,53 +18,56 @@
+						<rect class="message-placeholder-line-one" />
+						<rect class="message-placeholder-line-two" />
+						<rect class="message-placeholder-line-three" />
+						<rect class="message-placeholder-line-four" :style="{width: `width`}" />
```
The last line doesn't seem to have a random length anymore.

Something like the following makes it work:
```diff
diff --git a/src/components/LoadingPlaceholder.vue b/src/components/LoadingPlaceholder.vue
index 49acbfc0e..e74f28c58 100644
--- a/src/components/LoadingPlaceholder.vue
+++ b/src/components/LoadingPlaceholder.vue
@@ -51,7 +51,7 @@ each other by animating the opacities.
                                                :fill="'url(#placeholder-gradient' + suffix + ')'">
                                                <circle class="conversation-placeholder-icon" />
                                                <rect class="conversation-placeholder-line-one" />
-                                               <rect class="conversation-placeholder-line-two" :style="{width: `width`}" />
+                                               <rect class="conversation-placeholder-line-two" :style="width" />
                                        </svg>
                                        <svg
                                                v-if="type === 'messages'"
@@ -62,7 +62,7 @@ each other by animating the opacities.
                                                <rect class="message-placeholder-line-one" />
                                                <rect class="message-placeholder-line-two" />
                                                <rect class="message-placeholder-line-three" />
-                                               <rect class="message-placeholder-line-four" :style="{width: `width`}" />
+                                               <rect class="message-placeholder-line-four" :style="width" />
                                        </svg>
                                </li>
                        </ul>
@@ -97,7 +97,7 @@ export default {
                        const data = []
                        for (let i = 0; i < this.count; i++) {
                                // generate random widths
-                               data.push(Math.floor(Math.random() * 20) + 30)
+                               data.push('width: ' + (Math.floor(Math.random() * 20) + 30) + '%')
                        }
                        return data
                },
```

PVince81 replied:
Nice catch. Despite my compare the old and new this didn't catch my eye.

Tested and pushed.
